main.cc

```diff
@@ -524,6 +524,26 @@ sharded<locator::snitch_ptr> *the_snitch;
 sharded<service::storage_proxy> *the_storage_proxy;
 }
 
+static void initialize_local_info(sharded<db::system_keyspace>& sys_ks,
```

Comment:
Please, note, that this function implicitly "knows" that it's run in async context. This is a time-bomb, because ~~when~~ if we do anything with the way it's called it will take extra eye+brain cycles to understand it. It's worth either making it a coroutine right at once, or at least add a comment that it's run in async context. Can be done as a followup if this PR is going to get merged the way it is now

Reply:
Good point, no problem to turn it into coroutine in this PR, I'll do this.

Reply:
Initialization is unlikely to be coroutinized as a whole. It's better to use seastar threads for that because it's easier to debug. With threads it's possible to obtain full backtrace, and we can see where the main thread is blocked.

Reply:
Agree. It's enough to append _thread to the name to clearly mark it.

Reply:
ok, will add `_thread` suffix